### PR TITLE
Added Laravel Blade

### PR DESCRIPTION
**Pull request by jbrooksuk**
Ping @Alhadis.

### CODE REVIEW

**Comment by Alhadis** [comment]
Yup, looks good to me. =)

**Comment by jbrooksuk** [comment]
Sorry for being a complete idiot with this :)

**Comment by Alhadis** [comment]
Not your fault, mate. The whole submodule thing confuses me as well.

**Comment by jbrooksuk** [comment]
Travis has broken?

**Comment by Alhadis** [comment]
Travis has issues.

**Comment by Ingramz** [comment]
Due to certain design choices in Atom, I will have to rename `text.html.php.blade` to `text.html.blade`, which is more inline with other grammars building upon html/php. Even though this has 0 impact on the functionality of linguist, I would advise to not merge this before update, which I am working on at the moment. After I have released the new version, please update the references.

Sorry for the inconvenience caused.

**Comment by Alhadis** [comment]
> so please ignore the message above and feel free to merge it

Consider whatever you just said ignored! :D

[Unrelated](https://www.youtube.com/watch?v=mmejmx_MgQs).

**Comment by Alhadis** [comment]
@jbrooksuk Try pushing a meaningless change. It'll force Travis to restart - basically a bit of a "turn it off and on again" manoeuvre.

**Comment by jbrooksuk** [comment]
Done! :)

**Comment by Alhadis** [comment]
\*holds a gun to Travis's forehead* Now, be a good boy and don't do anything I won't like...

**Comment by jbrooksuk** [comment]
Failed 😢

**Comment by Alhadis** [comment]
Holy shit, I even warned the guy. __\*pulls trigger*__
Die, Travis, Die

Found the culprit, though, I'll point it out for you.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -330,6 +330,15 @@ BitBake:
   - .bb
   ace_mode: text
 
+Blade:
+  type: markup
+  group: HTML
+  extensions:
+    - .blade
+    - .blade.php
+  tm_scope: text.html.blade.php
```
Comment (Alhadis) [inline lib/linguist/languages.yml]: This needs to be `text.html.php.blade`. Wherever the scope is mentioned/listed, it needs to match what's listed in the [grammar file](https://github.com/jawee/language-blade/blob/master/grammars/blade.cson#L1). Verbatim.

**Comment by jbrooksuk** [comment]
Thanks! Pushed.

**Comment by Alhadis** [comment]
To be fair, that was a pretty easy mistake to make...

**Comment by Alhadis** [comment]
@pchaigno `script/licensed verify` is exiting with an error code of 1 because the grammar's license contains some additional notes about using parts of `language-php`. Does it need review, or would it be fine to simply change the license from `other` to `mit` in the actual license file?

Also, why the heck is it choking on Lean? James didn't even touch that part...

**Comment by Alhadis** [comment]
Wait, found out why Lean is throwing an error. It was [recently removed](https://github.com/ajaxorg/ace/commit/96e0aba61b8aa4b111e2c90cd905aa0c9aad7c5d) from Ace's list of supported languages (for undocumented/undisclosed reasons).

@jbrooksuk This isn't your fault, it's just crap timing... as usual. Could you change [this line](https://github.com/JamesForks/linguist/blob/0019f60ba7466b06922e904c929b64c837b7f450/lib/linguist/languages.yml#L1964) to be `ace_mode: text`?

**Comment by jbrooksuk** [comment]
Done! :)

**Comment by Alhadis** [comment]
Gotta hand it to ya mate, for all the hurdles and shit luck, you're being incredibly patient. =) (It's never easy the first time, but damn, it's not meant to be THIS hard)

**Comment by jbrooksuk** [comment]
Don't worry about it, I fully appreciate stuff like this happens sometimes :)

File: vendor/licenses/grammar/language-blade.txt [hunk 2]
```diff
@@ -0,0 +1,60 @@
+---
+type: grammar
+name: language-blade
+license: other
```
Comment (Alhadis) [inline vendor/licenses/grammar/language-blade.txt]: I think you can go ahead and change `other` to `mit`, mate.

Comment (jbrooksuk) [inline vendor/licenses/grammar/language-blade.txt]: :+1:

**Comment by jbrooksuk** [comment]
Done! Fingers crossed.

**Comment by Alhadis** [comment]
When you cross your fingers, only do it on one hand. Doing it on both brings bad luck instead of good.

**Comment by Alhadis** [comment]
All is well, I see. =)

BTW, the reason the Ace mode was removed was due to an unclear CLA (ambiguous licensing). Just happened to come at a pretty unfortunate time, haha.

Well done, fella. @pchaigno, am I right to believe editing the grammar's license from `other` to `mit` won't cause any further issues?

**Comment by larsbrinkhoff** [comment]
I made a separate PR for the Lean thing: #3108.

**Comment by jbrooksuk** [comment]
🎉  I cannot wait to see GitHub highlighting Blade files.

**Comment by jbrooksuk** [comment]
Any movement on this? :)

**Comment by Alhadis** [comment]
@arfon Any chance we can get this merged before the next gem is cut?

**Comment by jbrooksuk** [comment]
🎉 🎉 🎉 🎉

**Comment by jbrooksuk** [comment]
@arfon how long does it usually take to see linguist updates reflect in highlighting on GitHub itself?

**Comment by arfon** [comment]
> @arfon how long does it usually take to see linguist updates reflect in highlighting on GitHub itself? I assume you guys will release a new version, then run GitHub from that?

I need to cut a new Linguist gem and deploy it on GitHub. I'm planning on doing this today...

**Comment by jbrooksuk** [comment]
Sweet! Would it be at all possible for you to let me know when this is done? It'd be great news to spread in the Laravel community :)

**Comment by Alhadis** [comment]
Just punch Linguist's <kbd>Watch</kbd> button up the top to get e-mail notifications when there're changes. You can unwatch it once a release is cut.

**Comment by jbrooksuk** [comment]
@Alhadis thanks, but I'm more interested in when GitHub updates to support Blade :)

**Pull request merged**